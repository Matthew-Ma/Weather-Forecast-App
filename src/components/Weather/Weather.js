import React from 'react';
import { useSelector } from 'react-redux';
import Forecast from '../Forecast/Forecast';
import { Container, Header, Grid } from 'semantic-ui-react';
import SearchBox from '../Search/SearchBox';
import Spinner from '../ui/Spinner/Spinner';

const Weather = () => {
    const { loading } = useSelector((state) => ({
        loading: state.app.isLoading,
    }));

    return (
        <Container>
            {loading && <Spinner />}
            <Grid
                textAlign="center"
                style={{ height: '100vh' }}
                verticalAlign="middle"
            >
                <Grid.Column>
                    <Header size="huge" color="teal" textAlign="center">
                        Weather Forecast Widget
                    </Header>
                    <SearchBox />
                    <Forecast />
                </Grid.Column>
            </Grid>
        </Container>
    );
};

export default Weather;
