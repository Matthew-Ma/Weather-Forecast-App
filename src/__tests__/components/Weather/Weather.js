import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Weather from '../../../components/Weather/Weather';

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));

describe('Weather component:', () => {
    jest.spyOn(window, 'alert').mockReturnValue();
    beforeEach(() => {
        useSelector.mockImplementation((selector) =>
            selector({
                app: { isLoading: true },
            }),
        );
    });
    afterEach(() => {
        useSelector.mockClear();
    });

    it('renders Weather', () => {
        const { container } = render(<Weather />);
        expect(container).toMatchSnapshot();
    });
});
