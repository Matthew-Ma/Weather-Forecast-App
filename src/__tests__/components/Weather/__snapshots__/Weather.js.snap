// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component: renders Weather 1`] = `
<div>
  <div
    class="ui container"
  >
    <div
      class="loader"
    >
      <div
        class="sk-chase"
      >
        <div
          class="sk-chase-dot"
        />
        <div
          class="sk-chase-dot"
        />
        <div
          class="sk-chase-dot"
        />
        <div
          class="sk-chase-dot"
        />
        <div
          class="sk-chase-dot"
        />
        <div
          class="sk-chase-dot"
        />
      </div>
    </div>
    <div
      class="ui center aligned middle aligned grid"
      style="height: 100vh;"
    >
      <div
        class="column"
      >
        <div
          class="ui teal huge center aligned header"
        >
          Weather Forecast Widget
        </div>
        <div
          class="ui center aligned grid"
        >
          <div
            class="six wide column search-box"
          >
            <div
              class="ui big fluid icon input"
            >
              <input
                name="location"
                placeholder="Search for location"
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                class="search icon"
              />
            </div>
          </div>
        </div>
        <div
          class="ui four cards"
        />
      </div>
    </div>
  </div>
</div>
`;
